src/app/components/viewer/viewer.component.ts

```diff
@@ -131,8 +131,13 @@ export class AppViewerComponent implements OnInit, OnDestroy {
       .subscribe((ruleContext) => {
         this.selection = ruleContext.selection;
 
-        this.toolbarActions = this.extensions.getViewerToolbarActions();
-        this.openWith = this.extensions.openWithActions;
+        if (this.toolbarActions.length === 0) {
+          this.toolbarActions = this.extensions.getViewerToolbarActions();
```

Comment:
this may prevent evaluation of visibility rules

Reply:
is to avoid reload it after the first time, can they change after the first load?

Reply:
the `getViewerToolbarActions()` checks visibility conditions each time, maybe if you select folder of file, the buttons may change

Reply:
anyway, might need some checks if nothing gets brocken

Reply:
I did some checks didn't find a problem for now